File: dotnet-desktop-guide/winforms/migration/snippets/clipboard-dataobject-net10/net/vb/ModernApproach.vb
Imports System
Imports System.Windows.Forms

Namespace ClipboardExamples
    Public Class ModernApproach
        Public Class Person
            Public Property Name As String
            Public Property Age As Integer
        End Class

        ' <ModernTryGetData>
        Public Shared Sub ModernTryGetDataExample()
            Dim data As New Person With {.Name = "Alice", .Age = 30}
            Clipboard.SetDataAsJson("MyAppData", data)

            ' Use this - type-safe approach with TryGetData(Of T)()
            Dim person As Person = Nothing
            If Clipboard.TryGetData("MyApp.Person", person) Then
                ' person is guaranteed to be the correct type
                Console.WriteLine($"Processing person: {person.Name}, Age: {person.Age}")
            Else
                ' Handle the case where data isn't available or is the wrong type
                MessageBox.Show("Unable to retrieve person data from clipboard")
            End If
        End Sub
        ' </ModernTryGetData>
    End Class
End Namespace

File: dotnet-desktop-guide/winforms/migration/snippets/clipboard-dataobject-net10/net/csharp/ModernApproach.cs
using System;
using System.Windows.Forms;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace ClipboardExamples
{
    public class ModernApproach
    {
        public class Person
        {
            public string Name { get; set; }
            public int Age { get; set; }
        }

        // <ModernTryGetData>
        public static void ModernTryGetDataExample()
        {
            var data = new Person { Name = "Alice", Age = 28 };
            Clipboard.SetDataAsJson("MyAppData", data);

            // Use this - type-safe approach with TryGetData<T>()
            if (Clipboard.TryGetData("MyApp.Person", out Person person))
            {
                // person is guaranteed to be the correct type
                Console.WriteLine($"Processing person: {person.Name}, Age: {person.Age}");
            }
            else
            {
                // Handle the case where data isn't available or is the wrong type
                MessageBox.Show("Unable to retrieve person data from clipboard");
            }
        }
        // </ModernTryGetData>
    }
}
